src/Task/ConvertTranslatableTask.php

```diff
@@ -0,0 +1,209 @@
+                            $instanceLocale = SQLSelect::create()
+                                ->setFrom("\"{$baseTable}\"")
+                                ->setSelect('"Locale"')
+                                ->setWhere(["\"{$baseTable}\".\"ID\"" => $instance->ID])
+                                ->execute()
+                                ->first();
+
+                            // Ensure that we got the Locale out of the base table before continuing
+                            if (empty($instanceLocale['Locale'])) {
+                                Debug::message("Skipping {$instance->Title} with ID {$instanceID} - couldn't find Locale");
+                                continue;
+                            }
+                            $instanceLocale = $instanceLocale['Locale'];
+
+                            // Check for obsolete classes that don't need to be handled any more
+                            if ($instance->ObsoleteClassName) {
+                                Debug::message(
+                                    "Skipping {$instance->ClassName} with ID {$instanceID} because it from an obsolete class",
+                                    false
+                                );
+                                continue;
+                            }
+
+                            Debug::message(
+                                "Updating {$instance->ClassName} {$instance->Title} ({$instance->ID}) with locale {$instanceLocale}",
+                                false
+                            );
+
+                            FluentState::singleton()
+                                ->withState(function (FluentState $state) use ($instance, $instanceLocale) {
+                                    // Use Fluent's ORM to write and/or publish the record into the correct locale
+                                    // from Translatable
+                                    $state->setLocale($instanceLocale);
+
+                                    if (!$this->isPublished($instance)) {
+                                        $instance->write();
+                                        Debug::message("  --  Saved to draft", false);
+                                    } elseif ($instance->publishRecursive() === false) {
+                                        Debug::message("  --  Publishing FAILED", false);
+                                        throw new Exception("Failed to publish");
+                                    } else {
+                                        Debug::message("  --  Published", false);
+                                    }
+                                });
+                        }
+
+                        // Drop the "Locale" column from the base table
+                        Debug::message('Dropping "Locale" column from ' . $baseTable, false);
+                        DB::query(sprintf('ALTER TABLE "%s" DROP COLUMN "Locale"', Convert::raw2sql($baseTable)));
```

Comment:
Don't raw2sql on table names.

Reply:
It doesn't work. You can't put quotes / slashes etc into tables. raw2sql() only works on values not identifiers. In other words, it's the wrong escape type.

Reply:
Right, designed for values not identifiers. That makes sense. Thank you :)